pacman/operations/rigged_algorithms/random_placer.py

```diff
@@ -0,0 +1,97 @@
+from pacman.utilities.utility_objs import ResourceTracker
+from pacman.utilities.algorithm_utilities import placer_algorithm_utilities
+from pacman.model.placements import Placement, Placements
+from pacman.model.constraints.placer_constraints import SameChipAsConstraint
+
+from spinn_utilities.progress_bar import ProgressBar
+
+import random as default_random
+
+
+class RandomPlacer(object):
+
+    def __call__(self, machine_graph, machine):
+
+        # check that the algorithm can handle the constraints
+        ResourceTracker.check_constraints(machine_graph.vertices)
+
+        placements_copy = Placements()
+        vertices = \
+            placer_algorithm_utilities.sort_vertices_by_known_constraints(
+                machine_graph.vertices)
+
+        # Iterate over vertices and generate placements
+        progress = ProgressBar(machine_graph.n_vertices,
+                               "Placing graph vertices")
+        resource_tracker = ResourceTracker(machine,
+                                           self._generate_random_chips(
+                                               machine))
+        vertices_on_same_chip = \
+            placer_algorithm_utilities.get_same_chip_vertex_groups(
+                    machine_graph.vertices)
+        all_vertices_placed = set()
```

Comment:
placed_vertices? over all_vertices_placed. its not got all to begin with, so ambigious